src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red'| 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ((a >= (b + c)) || (b >= (a + c)) || (c >= (a + b))
+    || (a <= 0 || b <= 0 || c <= 0)) {
+      throw new Error('error: sides 1, 2 and 3 cannot form a triangle');
+    }
+  }
```

Comment:
better to make 2 checks and throw 2 different errors

Reply:
Thanks, I changed to 2 checks